eskip/template_test.go

```diff
@@ -57,3 +61,101 @@ func TestTemplateGetter(t *testing.T) {
 		nil,
 	}})
 }
+
+func TestTemplateApplyRequestResponseContext(t *testing.T) {
+	for _, ti := range []struct {
+		name           string
+		template       string
+		context        *filtertest.Context
+		requestExpect  string
+		requestOk      bool
+		responseExpect string
+		responseOk     bool
+	}{{
+		"path params",
+		"hello ${p1} ${p2}",
+		&filtertest.Context{
+			FParams: map[string]string{
+				"p1": "path",
+				"p2": "params",
+			},
+		},
+		"hello path params",
+		true,
+		"hello path params",
+		true,
+	}, {
+		"state bag",
+		"hello ${string} ${int} ${int8} ${int16} ${int32} ${int64} ${uint} ${uint8} ${uint16} ${uint32} ${uint64} ${float32} ${float64}",
+		&filtertest.Context{
+			FStateBag: map[string]interface{}{
+				"string":  "s",
+				"int":     int(-1),
+				"int8":    int8(-1 << 7),
+				"int16":   int16(-1 << 15),
+				"int32":   int32(-1 << 31),
+				"int64":   int64(-1 << 63),
+				"uint":    uint(1),
+				"uint8":   uint8(1<<8 - 1),
+				"uint16":  uint16(1<<16 - 1),
+				"uint32":  uint32(1<<32 - 1),
+				"uint64":  uint64(1<<64 - 1),
+				"float32": float32(2.7182818284_5904523536_0287471352),
+				"float64": float64(3.1415926535_8979323846_2643383279),
+			},
+		},
+		"hello s -1 -128 -32768 -2147483648 -9223372036854775808 1 255 65535 4294967295 18446744073709551615 2.7182817 3.141592653589793",
```

Comment:
https://stackoverflow.com/questions/28045787/how-many-decimal-places-does-the-primitive-float-and-double-support